comp/core/config/params.go

```diff
@@ -35,6 +35,10 @@ type Params struct {
 	// file does not exist.
 	configMissingOK bool
 
+	// configInvalidOK determines whether it is OK if the config is not valid
+	// If an error occurs, Component.warnings.Warning contains the error.
+	configInvalidOK bool
```

Comment:
I think this catches all error not just if the config is not valid. WDYT about renaming it `ignoreErrors` or something similar (and then update the `WithConfigInvalidOK` accordingly) ?

Reply:
Code updated.